src/main/kotlin/com/exactpro/th2/codec/fixng/ByteBufUtil.kt

```diff
@@ -90,16 +111,24 @@ inline fun ByteBuf.forEachField(
     charset: Charset,
     isDirty: Boolean,
     action: (tag: Int, value: String) -> Boolean,
+    warningHandler: (String) -> Unit,
 ) {
     while (isReadable) {
         val offset = readerIndex()
-        if (action(readTag(), readValue(delimiter, charset, isDirty))) continue
+        if (action(readTag(warningHandler), readValue(delimiter, charset, isDirty))) continue
         readerIndex(offset)
         break
     }
 }
 
-inline fun ByteBuf.readField(tag: Int, delimiter: Byte, charset: Charset, isDirty: Boolean, message: (Int) -> String): String = readTag().let {
+inline fun ByteBuf.readField(
+    tag: Int,
+    delimiter: Byte,
+    charset: Charset,
+    isDirty: Boolean,
+    message: (Int) -> String,
+    warningHandler: (String) -> Unit,
```

Comment:
suggestion: most of the time, the `warningHandler` is a small lambda expression. I think it would be better to keep `message` lambda as the last argument in the method and move `warningHandler` before it